lib/drivers/node-mongodb-native/connection.js

```diff
@@ -409,6 +415,12 @@ function _setClient(conn, client, options, dbName) {
       }
     });
   }
+  client.on('serverHeartbeatSucceeded', () => {
+    conn._lastHeartbeatAt = Date.now();
+  });
+  client.on('serverHeartbeatFailed', () => {
+    conn._lastHeartbeatAt = Date.now();
+  });
```

Comment:
should this really count as a heartbeat?

Reply:
It shouldn't matter because heartbeat failed would make Mongoose readyState `disconnected`. Which is a good argument for removing this code, because it doesn't have any impact. Thanks for the suggestion :+1: